salt/syspaths.py

```diff
@@ -32,14 +37,19 @@
 except ImportError:
     import types
     __generated_syspaths = types.ModuleType(str('salt._syspaths'))  # future lint: blacklisted-function
-    for key in ('ROOT_DIR', 'CONFIG_DIR', 'CACHE_DIR', 'SOCK_DIR',
-                'SRV_ROOT_DIR', 'BASE_FILE_ROOTS_DIR', 'HOME_DIR',
-                'BASE_PILLAR_ROOTS_DIR', 'BASE_THORIUM_ROOTS_DIR',
-                'BASE_MASTER_ROOTS_DIR', 'LOGS_DIR', 'PIDFILE_DIR',
-                'SPM_PARENT_PATH', 'SPM_FORMULA_PATH',
-                'SPM_PILLAR_PATH', 'SPM_REACTOR_PATH', 'SHARE_DIR'):
+    for key in svl:
         setattr(__generated_syspaths, key, None)
-
+else:
+    for key in svl:
+        if hasattr(__generated_syspaths, key):
+            continue
+        else:
+            if not typo_warning:
+                log.warning('Possible Typo?')
+                log.warning('To dissolve this warning add `[variable] = None` to _syspaths.py')
```

Comment:
Was your thought here that we should only show this typo warning if there's more than one missing value? It seems somewhat odd to me that we'd show it for subsequent errors and not for the first (after all, the first one *could* be a typo).

I think I'd either display it for all or for none - though I'd probably just include it in the other log message.